documentation/writing-installers.md

```diff
@@ -0,0 +1,100 @@
+# Writing Installers
+
+Generate the install task:
+```
+mix igniter.gen.task my_dep.install
+```
+All of the work will be done in the igniter call:
+
+```elixir
+@impl Igniter.Mix.Task
+def igniter(igniter) do
+  # Do your work here and return an updated igniter
+  igniter
+  |> Igniter.add_warning("mix routex.install is not yet implemented")
+end
+```
+
+## Fetching Dependency
+
+```elixir
+if Igniter.Project.Deps.has_dep?(igniter, :my_dep) do
```

Comment:
So you don't have to do this typically. I'd say let's remove this example. If your installer should install additional packages or add additional dependencies, that should be configured in the installer. i.e `installs: [{:other_dep, ...}], adds_deps: [{:other_dep2, ...}]`.

You don't need to add the specific dep that you are installing, as igniter handles that.

Reply:
🤔 
When I comment out `fetch_dep` which does above and do in info:
```
        adds_deps: [{:routex, "~> 1.1"}],
        installs: [{:routex, "~> 1.1"}],
```
No changes detected on `assert_has_patch("mix.exs", ...)`

Refering to [this](https://github.com/BartOtten/routex/pull/82)

Am I missing something? :thinkies:

Reply:
In tests we use `compose_task` which does not call the full installer. but when using `igniter.install` it will.